examples/erc20/DAI/DAI.xml

```diff
@@ -61,18 +94,192 @@ xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            <ts:label>
+                <ts:string xml:lang="en">Repay AAVE Debt</ts:string>
+            </ts:label>
+            <ts:attribute name="repayAmount">
+                <ts:type><ts:syntax>1.3.6.1.4.1.1466.115.121.1.36</ts:syntax></ts:type>
+                <ts:label>
+                    <ts:string xml:lang="en">Amount to Repay</ts:string>
+                </ts:label>
+                <ts:origins>
+                    <!-- e18 is a hard coded multiplier.
+                    rationale for hardcoding: avoiding over-design  -->
+                    <ts:user-entry as="e18"/>
+                </ts:origins>
+            </ts:attribute>
+            <!-- default to user, unless they explicitly write someone else's address -->
+            <ts:attribute name="debtHolder">
+                <ts:type><ts:syntax>1.3.6.1.4.1.1466.115.121.1.36</ts:syntax></ts:type>
+                <ts:label>
+                    <ts:string xml:lang="en">the address of whom's debt to repay</ts:string>
+                </ts:label>
+                <ts:origins>
+                    <ts:user-entry as="address"/>
+                </ts:origins>
+            </ts:attribute>
+            <ts:transaction>
+                <ethereum:transaction function="repay" contract="LendingPoolProvider" as="uint">
                     <ts:data>
-                        <ts:address ref="ownerAddress"/>
+                        <ts:address>0x6b175474e89094c44da98b954eedeac495271d0f</ts:address>
+                        <ts:uint256 ref="repayAmount"/>
+                        <ts:address ref="debtHolder"/>
                     </ts:data>
-                </ethereum:call>
+                </ethereum:transaction>
+            </ts:transaction>
+            <ts:view xml:lang="en" xmlns="http://www.w3.org/1999/xhtml">
+                <xhtml:style type="text/css">&style;</xhtml:style>
+                <xhtml:script  type="text/javascript">&repay.en;</xhtml:script>
+            </ts:view>
+        </ts:card>
+
+        <ts:card type="activity" name="aETHRepaid">
+            <ts:origins>
+                <ethereum:event type="Repay" filter="ownerAddress=${_user}"/>
             </ts:origins>
-        </ts:attribute>
+            <ts:item-view xml:lang="en" xmlns="http://www.w3.org/1999/xhtml">
+                <xhtml:style type="text/css">&style;</xhtml:style>
+                <xhtml:script  type="text/javascript">&item-view-repay.en;</xhtml:script>
+            </ts:item-view>
+            <ts:view xml:lang="en" xmlns="http://www.w3.org/1999/xhtml">
+                <xhtml:style type="text/css">&style;</xhtml:style>
+                <xhtml:script  type="text/javascript">&item-view-repay.en;</xhtml:script>
+            </ts:view>
+        </ts:card>
+
+        <ts:card type="activity" name="aETHRepaidFromAnotherUser">
+            <ts:origins>
+                <ethereum:event type="Repay" filter="&amp;(ownerAddress=${_user}&amp;!(ownerAddress=${_repayer})"/>
```

Comment:
There's a syntax error in the filter, probably should be:

```
&amp;(ownerAddress=${_user}(!(ownerAddress=${_repayer})))
```

(I haven't tested it  yet)